bindings/node/lib/clientEncryption.js

```diff
@@ -202,12 +165,37 @@ module.exports = function(modules) {
      * });
      */
     createDataKey(provider, options, callback) {
-      if (typeof options === 'function') (callback = options), (options = {});
+      if (typeof options === 'function') {
+        callback = options;
+        options = {};
+      }
+      if (typeof options === 'undefined') {
+        options = {};
+      }
 
       const bson = this._bson;
-      options = sanitizeDataKeyOptions(bson, options);
-      const dataKeyBson = bson.serialize(Object.assign({ provider }, options.masterKey));
-      const context = this._mongoCrypt.makeDataKeyContext(dataKeyBson);
+
+      const dataKey = Object.assign({ provider }, options.masterKey);
+
+      let keyAltNames;
+      if (Array.isArray(options.keyAltNames) && options.keyAltNames.length > 0) {
```

Comment:
the check for the length makes it so that `keyAltNames` is `undefined` if an empty options array is passed in; is that the desired behavior?

Reply:
Yep, the c++ checks for undefined before using this option so I figure if there's nothing in the array there's nothing of relevance to pass in. It would be equivalent to pass down an empty array.